modules/fedimint-ln-client/src/lib.rs

```diff
@@ -274,10 +274,28 @@ impl LightningClientExt for Client {
             (PayType::Lightning(operation_id), output, contract_id)
         };
 
+        // TODO: return fee from create_outgoing_output or even let user supply
+        // it/bounds for it
+        let fee = match &output.output {
+            LightningOutput::Contract(contract) => {
+                let fee_msat = contract
+                    .amount
+                    .msats
+                    .checked_sub(
+                        invoice
+                            .amount_milli_satoshis()
+                            .ok_or(anyhow::anyhow!("MissingInvoiceAmount"))?,
+                    )
+                    .expect("Contract amount should be greater or equal than invoice amount");
```

Comment:
panic in a client? :thinking:

Reply:
Maybe better to just return an `anyhow::anyhow!` error and prevent the client from paying? The client probably should do some reasonable fee checking before attempting the payment

Reply:
This happening would indicate a bug in the client logic. Underfunding an outgoing contract doesn't make any sense, the gateway will just reject. So this really cannot happen unless we make a mistake (e.g. forgetting to change this should we ever implement negative fees).